docs/advanced/advanced_running_stateful_filesystem.md

```diff
@@ -0,0 +1,106 @@
+      serviceName: demo-statefulset
+      replicas: 1
+      template:
+        metadata:
+          labels:
+            app: demo-statefulset
+        spec:
+          containers:
+          - name: demo-container
+            image: registry.access.redhat.com/ubi8/ubi:latest
+            command: [ "/bin/sh", "-c", "--" ]
+            args: [ "while true; do sleep 30; done;" ]
+            <b>volumeMounts:
+              - name: demo-volume-file-system
+                mountPath: "data"</b>
+          volumes:
+          - name: demo-volume-file-system
+            persistentVolumeClaim:
+              claimName: demo-pvc-file-system
+      #      nodeSelector:
+      #        kubernetes.io/hostname: HOSTNAME
+3. Check the newly created pod.
+
+    Display the newly created pod (make sure the pod status is _Running_).
+
+    <pre>
+    $> kubectl get pod demo-statefulset-file-system-0
+    NAME                 READY   STATUS    RESTARTS   AGE
+    demo-statefulset-file-system-0   1/1     Running   0          43s
+
+4. Write data to the persistent volume of the pod.
+
+    The PV should be mounted inside the pod at `/data`.
+
+    <pre>
+    $> kubectl exec demo-statefulset-file-system-0 -- touch /data/FILE
+    $> kubectl exec demo-statefulset-file-system-0 -- ls /data/FILE
+    /data/FILE
+
+5. Delete StatefulSet and then recreate, in order to validate data (/data/FILE) remains in the persistent volume.
+
+    1. Delete the StatefulSet.
+
+        <pre>
+        $> kubectl delete statefulset/demo-statefulset-file-system
+        statefulset/demo-statefulset-file-system deleted
+
+    2. Wait until the pod is deleted. Once deleted, the `"demo-statefulset-file-system" not found` is returned.
+
+        <pre>
+        $> kubectl get statefulset/demo-statefulset-file-system
+        Error from server (NotFound): statefulsets.apps <StatefulSet name> not found
+
+    3. Verify that the multipath was deleted and that the PV mountpoint no longer exists by establishing an SSH connection and logging into the worker node.
+          
+        <pre>
+        $> ssh root@k8s-node1
+            
+        $>[k8s-node1] df | egrep pvc
+        $>[k8s-node1] multipath -ll
+        $>[k8s-node1] lsblk /dev/sdb /dev/sdc
+        lsblk: /dev/sdb: not a block device
+        lsblk: /dev/sdc: not a block device
+
+    4. Recreate the StatefulSet and verify that /data/FILE exists.
+
+        <pre>
+        $> kubectl create -f demo-statefulset-file-system.yaml
+        statefulset/demo-statefulset-file-system created
+            
+        $> kubectl exec demo-statefulset-file-system-0 -- ls /data/FILE
+        /data/FILE
+
+6. Delete StatefulSet and the PVC.
+
+    <pre>
+    $> kubectl delete statefulset/demo-statefulset-file-system
+    statefulset/demo-statefulset-file-system deleted
+        
+    $> kubectl get statefulset/demo-statefulset-file-system
+    Error from server (NotFound): statefulsets.apps <StatefulSet name> not found.
```

Comment:
I thought we have a different [placeholder format](https://github.com/IBM/ibm-block-csi-driver/blob/bd199c30f3801ac7bbef2293c18052a12bf90d0a/docs/content/configuration/csi_ug_config_create_secret.md#creating-an-array-secret-file) convention.
```suggestion
    Error from server (NotFound): statefulsets.apps <statefulset-name> not found.
```

Reply:
@oriyarde  I made it copyable and it is showing now.
what should "demo-statefulset-raw-block" be replaced with? just "statefulset-name"? isn't one the name and the other the name of the yaml file itself?
FYI @lisyukanna

Reply:
thanks!
in this case the name of the file is irrelevant.
unlike commands with `-f` (e.g. `apply -f`), `get` commands refer to names of objects on the server side (and the file is on the client side).

so if we want to use a placeholder, we should use it throughout this file.
if not, we should not use it at all in this file

Reply:
thanks, @oriyarde  I'll change all instances of `demo-statefulset-file-system` and `demo-statefulset-raw-block` with the general statefulset-name. I will look at the outcome after and see what I can do to streamline the information as much as possible.
Please confirm that this change also applies to the following instances: `demo-statefulset-file-system-0`

![image](https://user-images.githubusercontent.com/54570654/140642344-2eadadf1-4a4d-4e3f-b5c3-fe877344af95.png)

Reply:
thanks.
regarding `demo-statefulset-file-system-0` - I think it is the only name in our docs that is derived from user input but not identical (statefulset "adds" `-0` to its pod name).
we can go with a slightly confusing (but correct) `<statefulset-name>-0` or avoid the placeholder

Reply:
thanks.
I updated to statefulset-name-0 but without the < >. please review and let me know where these need to be added in so I can do it all at once. thanks

Reply:
I don't think it's correct to omit the `<>`, because it makes it hard to distinguish between the placeholder and the rest

Reply:
@oriyarde  i'll update but please review and let me know if there is anywhere else that will need to have these added in.

In addition - please see https://github.com/IBM/ibm-block-csi-driver/blob/docs_advanced_oct21/docs/advanced/advanced_running_stateful.md. It's a trial file to see if if works to have both files (filesystem and block) together

FYI @lisyukanna

Reply:
thanks, I'm not aware of other places where these need to be added in.

I like the single file approach - very nice!

Reply:
thanks. once the other comments for these files are closed i'll update the information and delete the separated files.

Reply:
great, so I'm resolving this, and if there would be any gaps later, I'll open another conversation